### ISSUE TITLE
Targeting specific `declaration-property-value-blacklist` values

### ISSUE THREAD

**Issue by Ne-Ne** [stylelint/stylelint#2534]
At the moment I want to target this last part of the value, however they are multiple iterations of the shorthand to cover this. While this is not a bug, it more of a question (possible feature request). I cannot find any documentation on the site to crack this nut..

The plugin works as expected, however I want stylelint to address this flex issue with the flex-basis value [flexbugs#4](https://github.com/philipwalton/flexbugs#4-flex-shorthand-declarations-with-unitless-flex-basis-values-are-ignored).

Targeting the last value in flex using `declaration-property-value-blacklist`. I want to fail linting when code with the last flex-basis value is set to 0, but only in flex shorthand (ref : https://github.com/stylelint/stylelint/issues/551)

> Which rule, if any, is this issue related to?

`declaration-property-value-blacklist`

> What CSS is needed to reproduce this issue?

```css
.foo {
  flex: 1 0 0;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
"declaration-property-value-blacklist": {
  "flex": [
    "1 0 0",
    "1 1 0",
    "0 1 0",
    "0 0 0"
    ]
},
```

> Which version of stylelint are you using?

`7.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`"gulp-stylelint": "3.7.0"`

Any help would be greatly appreciated!

**Comment by evilebottnawi** [comment]
@Ne-Ne why don't use regex (https://stylelint.io/user-guide/rules/declaration-property-value-blacklist/#options)?

**Comment by evilebottnawi** [comment]
` "flex": ["/(0|1) (0|1) 0/"],` ?

**Comment by Ne-Ne** [comment]
@evilebottnawi Thanks for the fast response

I need to exclude the percentage as that will validate fine in ie. I also need to provide a message to this flex property only.

**Comment by evilebottnawi** [comment]
@Ne-Ne get full list allowed and disallowed values, maybe i can help with this

**Comment by Ne-Ne** [comment]
Without creating many lines, this is quick mock up:

```
{
"declaration-property-value-blacklist": {
  "flex": [
    "/(0|1) (0|1) (not["0em", "0px", "0vw", "0vmin", "0vmax", "0vh", "0ch", "0ex", "0pc"], allow ["0%"])/"
    ]
},
```

**Comment by evilebottnawi** [comment]
@Ne-Ne difficult to understand can you be more clarify:
Bad:
```shell
flex: 1 0 0;
...
```
Good:
```shell
flex: 1 100%;
...
```

**Comment by Ne-Ne** [comment]
That is correct, the flex-basis can't be 0

It could be auto or {value}{unit} not empty either.

**Comment by evilebottnawi** [comment]
@Ne-Ne `declaration-property-value-blacklist` rule not for validate values for properties use `https://www.npmjs.com/package/stylelint-csstree-validator` for this

**Comment by Ne-Ne** [comment]
It turns out @evilebottnawi this was more than enough:

> "flex": ["/(0|1) (0|1) 0/"]
